core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This is implementation for Long Running Operations.
+ **/
+public class Poller<T> {
+
+    /*pollOperation is a Function that takes the previous PollResponse, and
+    returns a new PollResponse to represent the current state*/
+    private Function<PollResponse<T>, PollResponse<T>> pollOperation;
+
+    /*Various configuration options to create poller object.*/
+    private PollerOptions pollerOptions;
+
+    /*This will save last poll response.*/
+    private PollResponse<T> pollResponse;
+
+    /*This will be called when cancel operation is triggered.*/
+    private Consumer<Poller> cancelOperation;
+
+    /* Indicate to poll automatically or not when poller is created.
+     * default value is false;*/
+    private boolean autoPolling = true;
+
+    /*Indicate if cancel is initiated.*/
+    private boolean cancelInitiated;
+
+    private Flux<PollResponse<T>> fluxHandle;
+
+    /*Since constructor create a subscriber and start autopoll.
+     * This subscriber will be duplicate when client call subscriber.
+     * Thus this handle will be used to dispose the subscriber when
+     * client invoke poll function*/
+    private Disposable fluxDisposable;
+
+    /**
+     * Create a Poller that is configured to auto-poll.
+     *
+     * @param pollerOptions configuration options for poller.
+     * @param pollOperation This is the operation to be called.
+     **/
+    public Poller(PollerOptions pollerOptions,
+                  Function<PollResponse<T>,
+                      PollResponse<T>> pollOperation) {
```

Comment:
Put the Function parameter on a single line to make it more easy to read. My initial reaction to this was that this constructor had three arguments and you hadn't aligned PollResponse correctly.